db/primary_migrate/20230113202352_create_sp_return_logs_returned_at_date_issuer_index.rb

```diff
@@ -0,0 +1,20 @@
+class CreateSpReturnLogsReturnedAtDateIssuerIndex < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def up
+    safety_assured do
+      execute <<-SQL
+        CREATE INDEX CONCURRENTLY index_sp_return_logs_on_returned_at_date_issuer
+        ON public.sp_return_logs
+        USING btree ((returned_at::date), issuer)
+        WHERE (billable = true);
```

Comment:
```suggestion
        WHERE (billable = true and returned_at IS NOT NULL);
```

This probably doesn't make much difference for newer rows, but may as well

Reply:
I took `IS NOT NULL` out of the queries and confirmed that NULLs are not included in BETWEEN statements

Reply:
Yeah, it would primarily be to reduce the size of the index. We don't create any new rows with NULL `returned_at`, but there are existing ones.

Reply:
ahhh I see, will add. thanks!

Reply:
I'm not 100% sure if the query will still use the index if it doesn't specify the same conditions (but it might?)

Reply:
I checked this locally (same index as PR but different amount of data) and the query in the DailyAuthsReport seems to use the index as expected

```
=> [{"QUERY PLAN"=>"GroupAggregate  (cost=33.28..33.34 rows=2 width=140)"},
 {"QUERY PLAN"=>"  Group Key: sp_return_logs.ial, sp_return_logs.issuer, service_providers.iaa"},
 {"QUERY PLAN"=>"  ->  Sort  (cost=33.28..33.29 rows=2 width=132)"},
 {"QUERY PLAN"=>"        Sort Key: sp_return_logs.ial, sp_return_logs.issuer, service_providers.iaa"},
 {"QUERY PLAN"=>"        ->  Nested Loop Left Join  (cost=20.77..33.27 rows=2 width=132)"},
 {"QUERY PLAN"=>"              ->  Hash Right Join  (cost=20.62..32.14 rows=2 width=104)"},
 {"QUERY PLAN"=>"                    Hash Cond: ((service_providers.issuer)::text = (sp_return_logs.issuer)::text)"},
 {"QUERY PLAN"=>"                    ->  Seq Scan on service_providers  (cost=0.00..11.10 rows=110 width=100)"},
 {"QUERY PLAN"=>"                    ->  Hash  (cost=20.59..20.59 rows=2 width=36)"},
 {"QUERY PLAN"=>"                          ->  Bitmap Heap Scan on sp_return_logs  (cost=4.13..20.59 rows=2 width=36)"},
 {"QUERY PLAN"=>"                                Recheck Cond: (billable AND (returned_at IS NOT NULL))"},
 {"QUERY PLAN"=>"                                Filter: (((requested_at)::date >= '2023-01-08'::date) AND ((requested_at)::date <= '2023-01-18'::date))"},
 {"QUERY PLAN"=>"                                ->  Bitmap Index Scan on index_sp_return_logs_on_returned_at_date_issuer  (cost=0.00..4.13 rows=323 width=0)"},
 {"QUERY PLAN"=>"              ->  Index Scan using agencies_pkey on agencies  (cost=0.15..0.57 rows=1 width=40)"},
 {"QUERY PLAN"=>"                    Index Cond: (id = service_providers.agency_id)"}]
```